pkg/plugin/connector/connutils/schema.go

```diff
@@ -0,0 +1,96 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connutils
+
+import (
+	"context"
+
+	"github.com/conduitio/conduit-connector-protocol/pconduit"
+	conduitschemaregistry "github.com/conduitio/conduit-schema-registry"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/conduitio/conduit/pkg/schemaregistry"
+	"github.com/conduitio/conduit/pkg/schemaregistry/fromschema"
+	"github.com/conduitio/conduit/pkg/schemaregistry/toschema"
+	"github.com/twmb/franz-go/pkg/sr"
+)
+
+type SchemaService struct {
```

Comment:
Btw, notice I moved `schemaregistry.Service` to `pkg/plugin/connector/connutils`, as this service should only be used by connectors. We will probably have something similar for processors. Now we have a central place to put future connector utilities (`StorageService` for instance).

It's similar to what you created before my refactoring, only that this service is used both for standalone and builtin plugins. This is also the type where we can add the connector token checks.